### ISSUE TITLE
nofib support

### ISSUE THREAD

**Issue by alpmestan** [snowleopard/hadrian#594]
Andreas (in [#15133](https://ghc.haskell.org/trac/ghc/ticket/15133)) and @tdammers (on the #ghc IRC channel) reported the problems they encountered while trying to run nofib on a GHC built by hadrian.

By default, nofib expects the usual inplace + ghc-stage<N> setup. We can point it to a separate GHC binary with commands like the ones I gave in #15133 (tl;dr: override `WithNofibHc` when running make commands).

Do we perhaps want to make this simpler and set those variables on behalf of hadrian users through dedicated `nofib` rule perhaps?

(Note: my "trick" does not allow me to complete the nofib run, see my first comment on the ticket mentionned above.)

**Comment by snowleopard** [comment]
@alpmestan Thanks, I like the idea of adding the `nofib` rule to Hadrian. Will it simply call the `Make` builder in the `nofib` directory with appropriate arguments? This sounds like it should be easy to implement.

**Comment by alpmestan** [comment]
@snowleopard Yes, that would/will in fact be very similar to how we run the tests.

**Comment by snowleopard** [comment]
@alpmestan OK, I've labelled this as `easy` :) I'm unlikely to find time for this in the coming two weeks as I am travelling.

If anyone would like to pick this up, please add a comment.

**Comment by alpmestan** [comment]
I took a quick stab at it, some further work is needed but it's a good start: https://github.com/snowleopard/hadrian/pull/599
